### ISSUE TITLE
feature importance using Captum 

### ISSUE THREAD

**Issue by dhaneeka** [bowang-lab/AGILE#18]
Hello,

According to the AGILE paper, Captum has been used to get the feature importance. In the code I could not locate the usage of Captum. Can you explain how you used Captum in the AGILE code, specifically in which script did you use it? I would be grateful if you can provide that part of the code within AGILE.

Thank you!

**Comment by Tina-112** [comment]
Hej rexxxx1234,
Congratulations on your paper in Nature Communications!
I read your paper with great enthusiasm and am very happy to see someone working on predicting LNP/mTP properties using deep learning. I am currently looking into whether we can use your work for a similar task. In doing so, I stumbled upon the same problem as dhaneeka. How did you calculate the importance of the features in Fig. 5 a/b? I tried to use the Integrated Gradients package from Captum, but I can't get it to work even though I rewrote the forward function of agile-finetune.py. I am also wondering, how did you manage to compute it for just the molecular descriptors?
I would really appreciate the code or just some input from you.
All the best!
Tina

**Comment by rexxxx1234** [comment]
Hi @dhaneeka and @Tina-112,

Thanks for your interest in AGILE! The server where I store my code is currently down, but I can push another branch with the Captum code soon!

Best,
Rex

**Comment by Tina-112** [comment]
Hi @rexxxx1234, 
that would be awesome - I'm looking forward to it!
Best, Tina

**Comment by rexxxx1234** [comment]
Hi @Tina-112 and @dhaneeka,

Captum code has been added to a new branch, you can use the `infer_vis_captum.py` file to obtain the feature importance from a fine-tuned model. Let me know if you have trouble running it!

**Comment by dhaneeka** [comment]
Hello Rex Ma,

Thank you very much for responding to my message and adding the new code
for the Captum. I really appreciate it. I will try it and see.

Best of luck with your next projects!
Dhaneeka
